applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -4,74 +4,76 @@
 //   _|\_\_|  \__,_|\__|\___/ ____/
 //                   Multi-Physics
 //
-//  License:		 BSD License
-//					 Kratos default license: kratos/license.txt
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
 //
 //  Main authors:    Ruben Zorrilla
 //
 
-#include "navier_stokes_wall_condition.h"
+// System includes
+
+
+// External includes
+
+
+// Project includes
 #include "includes/checks.h"
 
+// Application includes
+#include "navier_stokes_wall_condition.h"
+#include "wall_laws/navier_slip_wall_law.h"
+
 namespace Kratos
 {
 
 ///@name Specialized implementation of VMS for functions that depend on TDim
 ///@{
 
-
-/**
- * @see NavierStokesWallCondition::EquationIdVector
- */
-template <>
-void NavierStokesWallCondition<2,2>::EquationIdVector(EquationIdVectorType& rResult,
-                                                      const ProcessInfo& rCurrentProcessInfo) const
+template<unsigned int TDim, unsigned int TNumNodes, class... TWallModel>
+void NavierStokesWallCondition<TDim,TNumNodes,TWallModel...>::EquationIdVector(
+    EquationIdVectorType& rResult,
+    const ProcessInfo& rCurrentProcessInfo) const
 {
-    const unsigned int NumNodes = 2;
-    const unsigned int LocalSize = 6;
-    unsigned int LocalIndex = 0;
-
-    if (rResult.size() != LocalSize)
+    if (rResult.size() != LocalSize) {
         rResult.resize(LocalSize, false);
+    }
 
-    for (unsigned int iNode = 0; iNode < NumNodes; ++iNode)
-    {
-        rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_X).EquationId();
-        rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_Y).EquationId();
-        rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(PRESSURE).EquationId();
+    IndexType local_index = 0;
+    for (const auto& r_node : GetGeometry()) {
+        rResult[local_index++] = r_node.GetDof(VELOCITY_X).EquationId();
+        rResult[local_index++] = r_node.GetDof(VELOCITY_Y).EquationId();
+        if constexpr (TDim == 3) {
+            rResult[local_index++] = r_node.GetDof(VELOCITY_Z).EquationId();
+        }
+        rResult[local_index++] = r_node.GetDof(PRESSURE).EquationId();
```

Comment:
May be we could use `GetDofPosition` in here?

Reply:
We can. Done.